File: 78_Sine_Wave/java/src/SineWave.java
/**
 * Sine Wave
 *
 * Based on the Sine Wave program here
 * https://github.com/coding-horror/basic-computer-games/blob/main/78%20Sine%20Wave/sinewave.bas
 *
 * Note:  The idea was to create a version of the 1970's Basic program in Java, without introducing
 *        new features - no additional text, error checking, etc has been added.
 */
public class SineWave {

    public static void main(String[] args) {
        System.out.println("""
           SINE WAVE
           CREATIVE COMPUTING  MORRISTOWN, NEW JERSEY
           """);
        var isCreative = true;
        for(var t = 0d; t<40; t += .25) {
            //Indent output
            var indentations = 26 + (int) (25 * Math.sin(t));
            System.out.print(" ".repeat(indentations));
            //Change output every iteration
            var word = isCreative ? "CREATIVE" : "COMPUTING";
            System.out.println(word);
            isCreative  = !isCreative ;
        }
    }
}

File: 78_Sine_Wave/lua/sinewave.lua
require("math")
require("string")

print("\n                         Sine Wave")
print("           Creative Computing Morriston, New Jersy")
print("\n\n\n\n")

-- Original BASIC version by David Ahl
-- Ported to lua by BeeverFeever(github), 2022

local toggleWord = true

for t = 0, 40, 0.25 do
    local gap = math.floor(26 + 25 * math.sin(t))
    if toggleWord == true then
        -- string.rep used to add the gat at the front of the printed out words
        print(string.rep(" ", math.floor(gap)) .. "Creative")
    elseif toggleWord == false then
        print(string.rep(" ", math.floor(gap)) .. "Computing")
    end
end
